File: frontend/lib/models/group.dart
class Group {
  final int id;
  final String name;
  final String addictionType;
  final String location;
  final int leaderId;

  Group({
    required this.id,
    required this.name,
    required this.addictionType,
    required this.location,
    required this.leaderId,
  });

  factory Group.fromJson(Map<String, dynamic> json) {
    return Group(
      id: json['ID'] as int,
      name: json['name'] as String,
      addictionType: json['addiction_type'] as String,
      location: json['location'] as String,
      leaderId: json['leader_id'] as int,
    );
  }
}

File: backend/models/group.go
package models

import "gorm.io/gorm"

type Group struct {
	gorm.Model
	Name          string `json:"name"`
	AddictionType string `json:"addiction_type"`
	Location      string `json:"location"`
	LeaderID      uint   `json:"leader_id"` // 0 = no leader yet
}
